src/scanpy/plotting/_dotplot.py

```diff
@@ -93,143 +96,108 @@ class DotPlot(BasePlot):
-    DEFAULT_PLOT_X_PADDING = 0.8  # a unit is the distance between two x-axis ticks
-    DEFAULT_PLOT_Y_PADDING = 1.0  # a unit is the distance between two y-axis ticks
-
-    @old_positionals(
-        "use_raw",
-        "log",
-        "num_categories",
-        "categories_order",
-        "title",
-        "figsize",
-        "gene_symbols",
-        "var_group_positions",
-        "var_group_labels",
-        "var_group_rotation",
-        "layer",
-        "expression_cutoff",
-        "mean_only_expressed",
-        "standard_scale",
-        "dot_color_df",
-        "dot_size_df",
-        "ax",
-        "vmin",
-        "vmax",
-        "vcenter",
-        "norm",
+    size_title: str = "Fraction of cells\nin group (%)"
+    color_legend_title: str = "Mean expression\nin group"
+    legends_width: float = 1.5  # inches
+    show_size_legend: bool = True
+    show_colorbar: bool = True
+
+    # deprecated default class variables
+    DEFAULT_COLOR_ON: ClassVar[DefaultProxy[Literal["dot", "square"]]] = DefaultProxy(
+        "color_on"
     )
-    def __init__(
-        self,
-        adata: AnnData,
-        var_names: _VarNames | Mapping[str, _VarNames],
-        groupby: str | Sequence[str],
-        *,
-        use_raw: bool | None = None,
-        log: bool = False,
-        num_categories: int = 7,
-        categories_order: Sequence[str] | None = None,
-        title: str | None = None,
-        figsize: tuple[float, float] | None = None,
-        gene_symbols: str | None = None,
-        var_group_positions: Sequence[tuple[int, int]] | None = None,
-        var_group_labels: Sequence[str] | None = None,
-        var_group_rotation: float | None = None,
-        layer: str | None = None,
-        expression_cutoff: float = 0.0,
-        mean_only_expressed: bool = False,
-        standard_scale: Literal["var", "group"] | None = None,
-        dot_color_df: pd.DataFrame | None = None,
-        dot_size_df: pd.DataFrame | None = None,
-        ax: _AxesSubplot | None = None,
-        vmin: float | None = None,
-        vmax: float | None = None,
-        vcenter: float | None = None,
-        norm: Normalize | None = None,
-        **kwds,
-    ) -> None:
-        BasePlot.__init__(
-            self,
-            adata,
-            var_names,
-            groupby,
-            use_raw=use_raw,
-            log=log,
-            num_categories=num_categories,
-            categories_order=categories_order,
-            title=title,
-            figsize=figsize,
-            gene_symbols=gene_symbols,
-            var_group_positions=var_group_positions,
-            var_group_labels=var_group_labels,
-            var_group_rotation=var_group_rotation,
-            layer=layer,
-            ax=ax,
-            vmin=vmin,
-            vmax=vmax,
-            vcenter=vcenter,
-            norm=norm,
-            **kwds,
-        )
+    DEFAULT_DOT_MAX: ClassVar[DefaultProxy[float | None]] = DefaultProxy("dot_max")
```

Comment:
Unfortunate that the types need to be duplicated i.e., `dot_max: float | None` and then here `DEFAULT_DOT_MAX: ClassVar[DefaultProxy[float | None]]`

Reply:
Maybe worth creating type variables?  Seems like that is more lines of code, though...